soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6324,38 +6152,43 @@ s32 func_8083E5A8(Player* this, GlobalContext* globalCtx) {
                     return 1;
                 }
 
-                func_8083E4C4(globalCtx, this, giEntry);
+                func_8083E4C4(globalCtx, this, &giEntry);
                 this->getItemId = GI_NONE;
             }
         } else if (CHECK_BTN_ALL(sControlInput->press.button, BTN_A) && !(this->stateFlags1 & PLAYER_STATE1_11) &&
                    !(this->stateFlags2 & PLAYER_STATE2_10)) {
             if (this->getItemId != GI_NONE) {
-                GetItemEntry* giEntry = &sGetItemTable[-this->getItemId - 1];
+                GetItemEntry giEntry;
+                if (this->getItemEntry.objectId == OBJECT_INVALID) {
+                    giEntry = ItemTable_Retrieve(-this->getItemId - 1);
+                } else {
+                    giEntry = this->getItemEntry;
+                }
                 EnBox* chest = (EnBox*)interactedActor;
                 if (CVar_GetS32("gFastChests", 0) != 0) {
-                    giEntry->gi = -1 * abs(giEntry->gi);
+                    giEntry.gi = -1 * abs(giEntry.gi);
                 }
 
-                if (giEntry->itemId != ITEM_NONE) {
-                    if (((Item_CheckObtainability(giEntry->itemId) == ITEM_NONE) && (giEntry->field & 0x40)) ||
-                        ((Item_CheckObtainability(giEntry->itemId) != ITEM_NONE) && (giEntry->field & 0x20))) {
+                if (giEntry.itemId != ITEM_NONE) {
+                    if (((Item_CheckObtainability(giEntry.itemId) == ITEM_NONE) && (giEntry.field & 0x40)) ||
+                        ((Item_CheckObtainability(giEntry.itemId) != ITEM_NONE) && (giEntry.field & 0x20))) {
                         this->getItemId = -GI_RUPEE_BLUE;
-                        giEntry = &sGetItemTable[GI_RUPEE_BLUE - 1];
+                        giEntry = ItemTable_Retrieve(GI_RUPEE_BLUE - 1);
```

Comment:
noting another `-1` in case that's a thing that we end up wanting to change

Reply:
Addressed by 69086f58c2b6bb00a273eefed4744ad456bba6ba